state/protocol/badger/state_test.go

```diff
@@ -169,88 +169,147 @@ func TestBootstrapAndOpen_EpochCommitted(t *testing.T) {
 	rootSnapshot := unittest.RootSnapshotFixture(unittest.CompleteIdentitySet())
 	epoch1FirstHeight := rootSnapshot.Encodable().Head.Height
 
-	t.Run("root snapshot", func(t *testing.T) {
+	// For the spork root snapshot, only the first height of the root epoch should be indexed.
+	// [x]
+	t.Run("spork root snapshot", func(t *testing.T) {
 		util.RunWithFollowerProtocolState(t, rootSnapshot, func(db *badger.DB, state *bprotocol.FollowerState) {
 			// first height of started current epoch should be known
 			firstHeight, err := state.Final().Epochs().Current().FirstHeight()
 			require.NoError(t, err)
 			assert.Equal(t, epoch1FirstHeight, firstHeight)
 			// final height of not completed current epoch should be unknown
 			_, err = state.Final().Epochs().Current().FinalHeight()
-			assert.ErrorIs(t, err, protocol.ErrEpochTransitionNotFinalized)
+			assert.ErrorIs(t, err, protocol.ErrUnknownEpochBoundary)
 		})
 	})
 
-	t.Run("with next epoch", func(t *testing.T) {
-		unittest.SkipUnless(t, unittest.TEST_TODO, "such behavior is not supported yet since SealingSegment should"+
-			"contain blocks with the same protocol state ID. For this test it means that blocks needs to be selected"+
-			"from the same epoch phase and cannot cross epoch boundaries")
+	// In this test we construct a snapshot where the sealing segment is entirely
+	// within a particular epoch (does not cross any boundary). In this case,
+	// no boundaries should be queriable in the API.
+	// [---<--->--]
+	t.Run("snapshot excludes start boundary", func(t *testing.T) {
+		var epochHeights *unittest.EpochHeights
 		after := snapshotAfter(t, rootSnapshot, func(state *bprotocol.FollowerState, mutableState protocol.MutableProtocolState) protocol.Snapshot {
 			builder := unittest.NewEpochBuilder(t, mutableState, state)
-			builder.BuildEpoch().CompleteEpoch()
-			heights, ok := builder.EpochHeights(1)
+			builder.BuildEpoch().
+				AddBlocksWithSeals(flow.DefaultTransactionExpiry, 1). // ensure sealing segment excludes start boundary
+				CompleteEpoch()                                       // building epoch 1 (prepare epoch 2)
+			var ok bool
+			epochHeights, ok = builder.EpochHeights(1)
 			require.True(t, ok)
-			return state.AtHeight(heights.Committed)
+			// return a snapshot with reference block in the Committed phase of Epoch 1
+			return state.AtHeight(epochHeights.CommittedFinal)
 		})
 
 		bootstrap(t, after, func(state *bprotocol.State, err error) {
 			require.NoError(t, err)
-			// first height of started current epoch should be known
-			firstHeight, err := state.Final().Epochs().Current().FirstHeight()
-			assert.Equal(t, epoch1FirstHeight, firstHeight)
-			require.NoError(t, err)
+			// first height of started current epoch should be unknown
+			_, err = state.Final().Epochs().Current().FirstHeight()
+			assert.ErrorIs(t, err, protocol.ErrUnknownEpochBoundary)
 			// final height of not completed current epoch should be unknown
```

Comment:
Lets add for sake of completeness queries to the previous epoch. Just to check the invariant that if first height of current epoch is not known then previous epoch data also should be not available.